pkg/apis/extensions/v1beta1/types.go

```diff
@@ -910,6 +910,9 @@ type ReplicaSetStatus struct {
 	// The number of pods that have labels matching the labels of the pod template of the replicaset.
 	FullyLabeledReplicas int32 `json:"fullyLabeledReplicas,omitempty" protobuf:"varint,2,opt,name=fullyLabeledReplicas"`
 
+	// The number of available replicas for this replica set.
+	AvailableReplicas int32 `json:"availableReplicas,omitempty" protobuf:"varint,4,opt,name=availableReplicas"`
```

Comment:
We use `AvailableReplicas` in Deployments. A pod is considered available once N seconds pass since it was transitioned to Ready (for d.spec.minReadySeconds=N). So, for Deployments with minReadySeconds=0, a pod is considered available once it is ready. @bgrant0607 do we want to document this in API conventions?

The field in the RC/RS would be useful: 1) on its own - currently just by looking at the status, users cannot determine if their apps are up and running, instead they need to look into pods. 2) Deployments already do this but they need to get into the pod level in order to determine pod availability, whereas for any other info they look into the replica set spec/status. Calculating available pods in the replica set level is pretty cheap and it will remove some complexity from the deployment controller (albeit not much until we consider adding minReadySeconds in RC/RS).